### PR TITLE
Setting

### PR DESCRIPTION
**Pull request by brunoLloret**
### Description
- Removed extra docker-compose file
- Removed script_language and shard_store tests due to API unfamiliarity
- Fixed DCO by amending commit with -s flag
- Rebased and squashed commits for easier DCO fix
- Ran linter with npm run lint --fix
- Ensured all remaining tests pass

Note to reviewers:
- DCO has been fixed
- Commits have been squashed
- Linter has been run
- All current tests are passing
- Remaining tests were already in tests/default directory (?)
- I will explore more the API to move ahead with more substantial tests, any hints or further indications are welcomed!

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by brunoLloret** [comment]
I will sign off the unsigned commit and resubmit

File: package.json [hunk 1]
```diff
@@ -63,5 +63,9 @@
     "ts-node": "^10.9.1",
     "typescript": "<5.4.0",
     "yaml": "^2.3.4"
+  },
+  "devDependencies": {
+    "dotenv": "^16.4.5",
+    "eslint-plugin-yaml": "^1.0.3"
```
Comment (dblock) [inline package.json:69]: This seems out of place. This project doesn't release a JavaScript library, so move these into `dependencies` to be consistent.

File: tests/default/_core/refresh.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
+
+description: Test the _refresh endpoint to ensure basic refresh functionality.
```
Comment (dblock) [inline tests/default/_core/refresh.yaml:3]: Super nit. We are testing the spec, not the functionality of the product. 

```suggestion
description: Test _refresh.
```

File: .cspell [hunk 3]
```diff
@@ -84,6 +84,7 @@ lucene
 Lucene
 lycheeverse
 marvinpinto
+memlock
```
Comment (dblock) [inline .cspell:87]: Are those used anywhere?

File: .gitignore [hunk 4]
```diff
@@ -1,21 +0,0 @@
...
-
-# Jekyll Website
-_site/
-
-# build output
-build/
-
-# coverage output
-/coverage/
-
```
Comment (dblock) [inline .gitignore:21]: Was this accidentally deleted? We want this.

File: eslint.config.mjs [hunk 5]
```diff
@@ -118,6 +118,7 @@ export default [
     },
     rules: {
       'yml/no-empty-document': 'off',
+      'yml/no-empty-mapping-value': 'off',
```
Comment (dblock) [inline eslint.config.mjs:121]: Why is this needed?

**Review by dblock — CHANGES_REQUESTED** [review]
Looks like a test failed, an API was likely introduced later than 1.3.7 and a test needs a `version: >= ...` tag. Make sure to confirm it's the case.

```
ERROR   _core/script_language.yaml (tests/default/_core/script_language.yaml)
    ERROR   CHAPTERS
        ERROR   Retrieve information about available script languages.
            PASSED  REQUEST BODY
            ERROR   RESPONSE STATUS (Expected status 200, but received 500: application/json. Internal Server Error)
            SKIPPED RESPONSE PAYLOAD BODY
            SKIPPED RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
```

I think you should make this PR just about the new tests and remove all the setup/package.json and related changes.

There's a lot of commits in this PR for what ends up being a small change, you should squash them into 1 commit in this PR to clean things up and make git history easier to read.

File: .cspell (hunk 3 above)
Comment (brunoLloret) [reply .cspell:87]: they were on my local docker-compose.yaml, got it!

File: eslint.config.mjs (hunk 5 above)
Comment (brunoLloret) [reply eslint.config.mjs:121]: linter was complaining when I left a value empty in one of the basic tests I was writing, I will delete it though and make sure it makes sense for future tests
